homeassistant/components/workday/__init__.py

```diff
@@ -1 +1,76 @@
 """Sensor to indicate whether the current day is a workday."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_per_platform
+
+from .const import CONF_ADD_HOLIDAYS, CONF_REMOVE_HOLIDAYS, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the workday component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Import configuration from sensor platform
+    config_platform = config_per_platform(config, "binary_sensor")
+    for p_type, p_config in config_platform:
+        if p_type != DOMAIN:
+            continue
```

Comment:
You can call `hass.config_entries.flow.async_init` in `async_setup_platform` (in binary_sensor.py) so you don't need to search the config of every HA installations even the ones without workday.

Add this to binary_sensor.py:
```python3
async def async_setup_platform(hass, config, add_entities, discovery_info=None) -> None:
    await hass.config_entries.flow.async_init(
                DOMAIN,
                context={"source": config_entries.SOURCE_IMPORT},
                data=config,
            )
```